mysql/README.md

```diff
@@ -4,42 +4,163 @@
+| <code style="word-break:break-all;">performance_schema_consumer_events_statements_history_long</code> | `ON` | Optional. Enables tracking of a larger number of recent queries across all threads. If enabled it increases the likelihood of capturing execution details from infrequent queries. |
+| `max_digest_length` | `4096` | Required for collection of larger queries. Increases the size of SQL digest text in `events_statements_*` tables. If left at the default value then queries longer than `1024` characters will not be collected. |
+| <code style="word-break:break-all;">`performance_schema_max_digest_length`</code> | `4096` | Must match `max_digest_length`. |
+| <code style="word-break:break-all;">`performance_schema_max_sql_text_length`</code> | `4096` | Must match `max_digest_length`. |
+
+<!-- xxz tab xxx -->
+<!-- xxz tabs xxx -->
+
+### Agent Database Access
+
+The Datadog Agent requires read-only access to the database in order to collect statistics and queries.
+
+The following instructions grant the agent permission to login from any host using `datadog@'%'`. This is required for managed databases like `Amazon RDS` or `Google CloudSQL` as it's not possible to install the datadog agent directly on the host. For self-hosted databases the datadog user can be restricted to be allowed to login only from localhost by using `datadog@'localhost'`. See the [MySQL documentation][5] for more info.
+
+<!-- xxx tabs xxx -->
+<!-- xxx tab "MySQL ≥ 8.0" xxx -->
+
+```SQL
+CREATE USER datadog@'%' IDENTIFIED WITH mysql_native_password by '<UNIQUEPASSWORD>';
+ALTER USER datadog@'%' WITH MAX_USER_CONNECTIONS 5;
+GRANT REPLICATION CLIENT ON *.* TO datadog@'%'
+GRANT PROCESS ON *.* TO datadog@'%';
+GRANT SELECT ON performance_schema.* TO datadog@'%';
 ```
 
-For mySQL 8.0+ create the `datadog` user with the native password hashing method:
+<!-- xxz tab xxx -->
+<!-- xxx tab "MySQL 5.6 & 5.7" xxx -->
 
-```shell
-mysql> CREATE USER 'datadog'@'localhost' IDENTIFIED WITH mysql_native_password by '<UNIQUEPASSWORD>';
-Query OK, 0 rows affected (0.00 sec)
+```SQL
+CREATE USER datadog@'%' IDENTIFIED BY '<UNIQUEPASSWORD>';
+GRANT REPLICATION CLIENT ON *.* TO datadog@'%' WITH MAX_USER_CONNECTIONS 5;
+GRANT PROCESS ON *.* TO datadog@'%';
+GRANT SELECT ON performance_schema.* TO datadog@'%';
 ```
 
-**Note**: `@'localhost'` is only for local connections. For remote connections, use the hostname/IP of your Agent. For more information, see the [MySQL documentation][5].
+<!-- xxz tab xxx -->
+<!-- xxz tabs xxx -->
 
-**Note**: If you encounter the following error message `(1045, u"Access denied for user 'datadog'@'127.0.0.1' (using password: YES)"))`, see the [MySQL Localhost Error documentation][18].
+The following schema and procedures are required for [Deep Database Monitoring](#deep-database-monitoring).
+
+```SQL
+CREATE SCHEMA IF NOT EXISTS datadog;
+GRANT EXECUTE ON datadog.* to datadog@'%'
```

Comment:
Missing semicolon at the end: `GRANT EXECUTE ON datadog.* to datadog@'%';`

Reply:
Fixed